src/common/libsubprocess/subprocess.c

```diff
@@ -79,6 +79,8 @@ struct subprocess_channel *channel_create (flux_subprocess_t *p,
     c->child_fd = -1;
     if (!(c->name = strdup (name)))
         goto error;
+    if ((c->buffer_space = cmd_option_bufsize (p, name)) < 0)
+        goto error;
```

Comment:
IMHO it would be clearer to initialize `c->buffer_space` in `channel_local_setup()` where the buffer is created, and then use `fbuf_space()` right after creation as the initial value.

It's strictly used for local subprocesses and initializing it here makes that less obvious.